run-directive/action.yml

```diff
@@ -0,0 +1,54 @@
+name: Run Directive
+run-name: test
+
+inputs:
+  hash:
+    description: 'Github commit hash to checkout'
+    required: false
+    default: ''
+  working-directory:
+    description: 'Name of component directory'
+    required: true
+  config:
+    description: 'Config JSON file'
+    required: true
+  stage:
+    description: 'Name of stage'
+    required: true
+  directive:
+    description: 'A directive to run. e.g. npm ci, npm run test'
+    required: true
+
+jobs:
+  run-directive:
+    name: Run Directive
+    using: "composite"
+
+    steps:
+      - name: Setup Environment
+        uses: Brightspace/bdp-ci-action/setup-env@master
+
+      - name: Checkout branch
+        uses: Brightspace/third-party-actions@actions/checkout
+        with:
+          ref: ${{ inputs.hash }}
+          fetch-depth: 0
+
+      - name: Use Node.js
+        uses: Brightspace/third-party-actions@actions/setup-node
+        with:
+          node-version-file: .nvmrc
+
+      - name: Add CodeArtifact npm registry
+        uses: Brightspace/codeartifact-actions/npm/add-registry@main
+        with:
+          auth-token: ${{secrets.CODEARTIFACT_AUTH_TOKEN}}
+
+      - name: Setup working directory
+        run: |
+          cp ${{ inputs.config }} ${{ inputs.working-directory }}/config.json
+          cd ${{ inputs.working-directory }}
```

Comment:
Minor: we can reduce the risk of script injection here by using [intermediate environment variables](https://docs.github.com/en/actions/security-for-github-actions/security-guides/security-hardening-for-github-actions#good-practices-for-mitigating-script-injection-attacks).